discovery_utils/enrichment/hansard.py

```diff
@@ -1,348 +1,478 @@
+        ml_labelstore = getter.get_labelstore(keywords=False)
+
+        # Switch to full mode if labelstores do not exist
+        if keyword_labelstore.empty and ml_labelstore.empty:
+            incremental = False
+
+        if incremental:
+            xml_files = getter.get_latest_debates()
+        else:
+            xml_files = getter.get_all_debates()
+
+        if not xml_files:
+            logger.info("No new debates to process. Exiting.")
+            return
 
-    if incremental:
-        keyword_labels = hansard_getter.get_labelstore(keywords=True)
-        ml_labels = hansard_getter.get_labelstore(keywords=False)
-        hansard_getter.get_latest_debates()
-    else:
-        keyword_labels = None
-        ml_labels = None
-        latest_debates = None
-        hansard_getter.get_all_debates()
+        # Process debates
+        logger.info("Processing debate files")
+        processed_debates = processor.process_debates(xml_files)
 
-    logger.info("Processing data...")
-    hansard_processor = HansardProcessor()
-    latest_debates = hansard_processor.process_debates()
+        # Enrich data
+        logger.info("Enriching debate data")
+        new_keyword_labels, new_ml_labels = enricher.enrich_data(
+            existing_debates, keyword_labelstore, ml_labelstore, processed_debates, incremental
+        )
 
-    logger.info("Enriching data...")
-    enricher = HansardEnricher(
-        # test_flag=True
-    )
+        # Append new debates to existing dataset and upload
+        enricher.append_and_upload_debates(existing_debates, processed_debates, reset)
 
-    logger.info("Appending and uploading Hansard debates")
-    combined_df = pd.concat([debates, latest_debates], ignore_index=True)
-    enricher.upload_parquet(combined_df)
+        # Generate and log enrichment report
+        report = enricher.generate_enrichment_report(new_keyword_labels, new_ml_labels)
+        logger.info(f"Enrichment report: {report}")
 
-    enricher.enrich_data(debates, keyword_labels, ml_labels, latest_debates, incremental)
+        logger.info("Hansard enrichment pipeline completed successfully")
 
-    logger.info("Hansard enrichment completed successfully.")
+    except ValueError as ve:
+        logger.error(f"Configuration error: {str(ve)}")
+        raise
+    except Exception as e:
+        logger.error(f"Error in Hansard enrichment pipeline: {str(e)}")
+        raise
 
 
 if __name__ == "__main__":
-    run_hansard_enrichment()
+    # Parse pipeline mode, default is incremental
+    parser = argparse.ArgumentParser(description="Run Hansard enrichment pipeline")
+    parser.add_argument("--full", action="store_true", help="Run full enrichment instead of incremental")
+    parser.add_argument("--test", action="store_true", help="Use a subset of the data for local testing")
+    parser.add_argument("--reset", action="store_true", help="Reset labelstores and run full enrichment")
```

Comment:
Great to have these options! What's the difference between "full" an "reset" - is it that if you run "full" it will append the labels to the label store instead of overwriting? 

But I guess if you're now deduplicating the label store, then it will delete the new labels anyways? So, do we need both 'full' and 'reset' options or maybe just 'full' is enough? Let me know though if I'm confused!

Reply:
Good point, I added the reset mode for development purposes to explicitly delete the label stores but on a full run they are overwritten anyway so having both modes is indeed redundant.